wikia/common/mw_database/dbconfig.py

```diff
@@ -0,0 +1,108 @@
+import collections
+import os
+import random
+import yaml
+
+
+ConnectionDetails = collections.namedtuple('ConnectionDetails', ['hostname', 'username', 'password', 'dbname'])
+
+
+class DatabaseConfig(object):
+    def __init__(self, config_file, connect_fn, service_name=None):
+        if config_file is None:
+            config_file = os.environ['WIKIA_DB_YML']
+        with open(config_file) as fp:
+            ds_conf = yaml.load(fp)
+        self.mw_config = ds_conf[0]
+        self.mw_config["templateOverridesByCluster"] = self.mw_config.get("templateOverridesByCluster", {})
+        self.mw_config["templateOverridesByServer"] = self.mw_config.get("templateOverridesByServer", {})
+        self.mw_config["templateOverridesByService"] = self.mw_config.get("templateOverridesByService", {})
+        self.cluster_config = ds_conf[1]
+        self.external_cluster_config = ds_conf[2]
+        self.connect_fn = connect_fn
+        self.service_name = service_name
+
+    def get_connection_details(self, dbname, master=False, wc_master=False, override_db_name=None, username=None,
```

Comment:
I was rather thinking about scenarios when you just added an entry to city_list to avoid weird errors when you are too fast. Do you think it's overkill?